readme.md

```diff
@@ -1,17 +1,17 @@
 
 ## Spec
 
-### `h(name, data?, children?) -> node`
+### `h(tag, data?, children?) -> node`
+
+1. `tag` must be type of `string` and can **NOT** carry extra data, like classes or ids (e.g. `div.foo` or `div#bar`)
+2. If the the 2nd parameter is true and the 3rd parameter is `undefined` the 2nd parameter becomes `data` or `children` respectively
```

Comment:
I don't think this works.  I remember trying various patterns like this in `h2ml` and the tests would fail.

You need to rely on the type checking of the next steps in order to properly determine the call signature:
> If 3rd param is `undefined` and 2nd param is true for steps 2 or 3, it becomes `data` or `children` repsectively